tools/benchmark/generate/main.go

```diff
@@ -131,6 +135,131 @@ var cratesioTop2000 = RebuildBenchmark{
+}
+
+var debianTop500 = RebuildBenchmark{
+	Filename: "debian_top_500.json",
+	Generator: func(ctx context.Context) (ps benchmark.PackageSet) {
+		var (
+			// The "all" arch is included in other arch indices.
+			archs      = []string{"amd64"}
+			popularity = "https://popcon.debian.org/by_inst"
+			repository = "https://deb.debian.org/debian"
+		)
+		resp, err := get(ctx, popularity)
+		if err != nil {
+			log.Fatalf("error fetching popularity: %v", err)
+		}
+		b := bufio.NewScanner(resp)
+		var popularPackages []string
+		for b.Scan() {
+			line := strings.TrimSpace(b.Text())
+			if len(line) == 0 || line[0] == '#' {
+				continue
+			}
+			f := strings.Fields(line)
+			if len(f) < 2 {
+				continue
+			}
+			popularPackages = append(popularPackages, f[1])
+		}
+		// Mapping from package name to a set of strings like "component/sourceName/version/artifact" where package name matches the artifact.
+		repo := map[string]map[string]bool{}
+		for _, arch := range archs {
+			index, err := get(ctx, repository+fmt.Sprintf("/indices/files/arch-%s.files", arch))
+			if err != nil {
+				log.Fatalf("error fetching arch %s index: %v", arch, err)
+			}
+			b = bufio.NewScanner(index)
+			for b.Scan() {
+				if !strings.HasPrefix(b.Text(), "./pool/") {
+					continue
+				}
+				line := strings.Split(strings.TrimPrefix(strings.TrimSpace(b.Text()), "./pool/"), "/")
+				if len(line) != 4 {
+					log.Fatalf("expected index line to be exactly 4 elements long: %s", line)
+				}
+				component := line[0]
+				// line[1] is the sourceName prefix directory and unimportant here.
+				sourceName := line[2]
+				artifact := line[3]
+				if !strings.HasSuffix(artifact, ".deb") {
+					continue
+				}
+				artifactPieces := strings.Split(artifact, "_")
+				if len(artifactPieces) != 3 {
+					log.Fatalf("Artifact name uses unexpected pattern. Expected three peices separated by '_', got: %s", artifact)
+				}
+				packageName := artifactPieces[0]
+				version := artifactPieces[1]
+				if _, ok := repo[packageName]; !ok {
+					repo[packageName] = map[string]bool{}
+				}
+				repo[packageName][fmt.Sprintf("%s/%s/%s/%s", component, sourceName, version, artifact)] = true
+			}
+		}
+		for _, p := range popularPackages[:500] {
+			if _, ok := repo[p]; !ok || len(repo[p]) == 0 {
+				log.Printf("Cannot find package %s in the indexes", p)
+			}
+			var component, sourceName string
+			var artifacts []struct{ Version, Artifact string }
+			// TODO: Optionally only include the artifacts provided by a specific release.
+			for a := range repo[p] {
+				pieces := strings.Split(a, "/")
+				if len(pieces) != 4 {
+					log.Fatalf("Unexpected artifact format %s", a)
+				}
+				if component == "" {
+					component = pieces[0]
+				}
+				if sourceName == "" {
+					sourceName = pieces[1]
+				}
+				if !strings.HasPrefix(a, component+"/"+sourceName) {
+					log.Printf("Package occured with different prefixes: %s (%s vs %s)", p, component+"/"+sourceName, a)
+					component = ""
```

Comment:
you may be able to use a goto here instead to jump down out of the loop. something like "goto next" where next is the bottom of the outer loop